nomenclature/codelist.py

```diff
@@ -193,7 +193,7 @@ def from_directory(cls, name: str, path: Path, file_glob_pattern: str = "**/*"):
         instance of cls (CodeList if not inherited)
 
         """
-        code_list: List[Code] = []
+        code_list: List[cls.code_basis] = []
```

Comment:
What you've written is not wrong but since all the special codes `RegionCode`, `VariableCode` inherit from `Code` anyway I would keep it like this, for readability reasons. 

```suggestion
        code_list: List[Code] = []
```

type hints in this case are purely for the developer anyway and I think it's more readable with `Code`.